.github/workflows/syncRootfs.ps1

```diff
@@ -0,0 +1,53 @@
+# Download latest LTS rootfs
+param(
+    [Parameter(Mandatory)][string]$Path
+)
+
+$rootfsDir = "${Path}"
+$sha256file = "${rootfsDir}\SHA256SUMS.txt"
+$tarball = "${rootfsDir}\jammy.tar.gz"
+
+if ( ! $(Test-Path -Path "${rootfsDir}") ) {
+    New-Item -Path "${rootfsDir}" -ItemType "directory" | Out-Null
+}
+
+# Testing if a cached rootfs exists
+$inCache = $False
+if ( $(Test-Path -Path "${sha256file}") -and $(Test-Path -Path "${tarball}") ) {
+    $inCache = $True
+}
+
+Invoke-WebRequest                                                       `
+    -Uri "https://cloud-images.ubuntu.com/wsl/jammy/current/SHA256SUMS" `
+    -OutFile "${sha256file}.tmp"
+
+if ( $inCache ) {
+    # Testing if the existing cache is up to date
+    $oldSha256 = $(Get-Content "${sha256file}")
+    $newSha256 = $(Get-Content "${sha256file}.tmp")
+
+    if ( "${oldSha256}" -eq "${newSha256}" ) {
+        Write-Output "Cache hit"
+        Remove-Item -Path "${sha256file}.tmp" 2>&1 | Out-Null
+        Exit(0)
+    }
+}
+
+# Missing or outdated cache: downloading rootfs
+Invoke-WebRequest                                                                                     `
```

Comment:
As you are downloading the sha256, maybe it’s worth checking that the rootfs which is downloaded is not corrupted too?

Reply:
I ended up making a big change: instead of relying on the old SHA256SUMS to validate the cache, I compute the hash of the existing tarball. This is then reused to validate the download.